pkg/reconciler/common/prune.go

```diff
@@ -122,9 +268,9 @@ func createCronJob(k kubernetes.Interface, ctx context.Context, pru v1alpha1.Pru
 		},
 	}
 
-	if _, err := k.BatchV1beta1().CronJobs(targetNs).Create(ctx, cj, v1.CreateOptions{}); err != nil {
+	if _, err := pruner.kc.BatchV1beta1().CronJobs(targetNs).Create(ctx, cj, v1.CreateOptions{}); err != nil {
 		if strings.Contains(err.Error(), "already exists") {
-			if _, err := k.BatchV1beta1().CronJobs(targetNs).Update(ctx, cj, v1.UpdateOptions{}); err != nil {
+			if _, err := pruner.kc.BatchV1beta1().CronJobs(targetNs).Update(ctx, cj, v1.UpdateOptions{}); err != nil {
```

Comment:
cron job are in v1 from `k8s v1.21` should we switch the api to v1?

Reply:
Yes, I am on a separate pr for this.